posthog/hogql/resolver.py

```diff
@@ -206,6 +207,13 @@ def visit_join_expr(self, node: ast.JoinExpr):
 
             if self.database.has_table(table_name):
                 database_table = self.database.get_table(table_name)
+
+                if isinstance(database_table, View):
+                    node.table = parse_select(str(database_table.query))
```

Comment:
if view, expand the query associated with view and parse as if if were a part of the overall query